google-cloud-firestore/lib/google/cloud/firestore/document_listener.rb

```diff
@@ -119,6 +123,81 @@ def stop
         def stopped?
           @listener.stopped?
         end
+
+        ##
+        # Register to be notified of errors when raised.
+        #
+        # If an unhandled error has occurred the listener will attempt to
+        # recover from the error and resume listening.
+        #
+        # Multiple error handlers can be added.
```

Comment:
FWIW, errors should be terminal conditions and I don't expect there to be any follow-up callbacks. Errors that are retried should ideally be hidden from the user.

Reply:
@dazuma Do you want to keep the ability to add multiple error callbacks to match `on_error` in other packages?

Reply:
@quartzmo Yes I think we should, for consistency.